src/tests/Application.vala

```diff
@@ -96,18 +96,38 @@ namespace Terminal.Test.Application {
         }
     }
 
+    public static bool has_display = true;
     public static int main (string[] args) {
         Intl.setlocale (LocaleCategory.ALL, "");
 
         GLib.Test.init (ref args);
 
+
         /* MainWindow always create a tab in home, make it the current dir
          * so that we don't create a extra tab during tests
          */
         Environment.set_current_dir (Environment.get_home_dir ());
 
+        GLib.Test.add_func ("/application/test-display", () => {
+            Terminal.Test.Application.has_display = true;
+            Gdk.set_allowed_backends ("x11");
+            var x11_display = Gdk.Display.open (null);
+            if (x11_display == null) {
+                Gdk.set_allowed_backends ("wayland");
```

Comment:
why testing for each back-end? We don't do anything with the fact that it is X/Wayland

Reply:
Perhaps I am misunderstanding something - I am trying to test whether there is *either* an X11 *or* a Wayland display.  If there is neither then tests will not be run.

Reply:
I don't think that this is necessary, doing `Gdk.Display.open (null) != null` should be enough I think